src/core/net/tcp6.cpp

```diff
@@ -221,6 +223,7 @@ Error Tcp::Endpoint::SendByExtension(size_t aNumBytes, uint32_t aFlags)
 
     if (IS_FASTOPEN(tp.t_flags))
     {
+        memset(&sin6p, 0, sizeof(sin6p));
```

Comment:
Should this be `ClearAllBytes()` too?

Reply:
Yes, that was the intention - for some reason this one was skipped. Now in the new commit.